backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/materialDispo/distribution.feature

```diff
@@ -301,3 +301,93 @@ Feature: create distribution to balance demand
+      | Identifier | MD_Candidate_Type | MD_Candidate_BusinessCase | M_Product_ID | DateProjected        | Qty | Qty_AvailableToPromise | M_Warehouse_ID | DD_Order_Candidate_ID |
+      | c_1        | DEMAND            | SHIPMENT                  | p_1          | 2022-07-04T00:00:00Z | 8   | -8                     | targetWH       |                       |
+      | c_2        | SUPPLY            | DISTRIBUTION              | p_1          | 2022-07-04T00:00:00Z | 0   | -8                     | targetWH       | c1                    |
+      | c_3        | DEMAND            | DISTRIBUTION              | p_1          | 2022-07-04T00:00:00Z | 0   | 0                      | sourceWH       | c1                    |
+      | c_4        | SUPPLY            | DISTRIBUTION              | p_1          | 2022-07-04T00:00:00Z | 8   | 0                      | targetWH       | c2                    |
+      | c_5        | DEMAND            | DISTRIBUTION              | p_1          | 2022-07-04T00:00:00Z | 8   | -8                     | sourceWH       | c2                    |
+
+
+# ###############################################################################################################################################
+# ###############################################################################################################################################
+# ###############################################################################################################################################
+# ###############################################################################################################################################
+# ###############################################################################################################################################
+# ###############################################################################################################################################
+  @from:cucumber
+  Scenario: DD_Order_Candidate + DD_Order is created to balance the full demand of the sales order
+    When update existing PP_Product_Plannings
+      | Identifier      | IsCreatePlan |
+      | productPlanning | Y            |
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID | DateOrdered | PreparationDate      | M_Warehouse_ID |
+      | SO         | true    | bpartner_1    | 2022-07-04  | 2022-07-04T00:00:00Z | targetWH       |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID | M_Product_ID | QtyEntered |
+      | ol_1       | SO         | p_1          | 14         |
+    And the order identified by SO is completed
+
+    Then after not more than 60s, the MD_Candidate table has only the following records
+      | Identifier | MD_Candidate_Type | MD_Candidate_BusinessCase | M_Product_ID | DateProjected        | Qty | Qty_AvailableToPromise | M_Warehouse_ID |
```

Comment:
how hard is to also check DD_Order_Candidate_ID and DD_OrderLine_ID?

Reply:
changed to check for them.